lib/rspec/support/differ.rb

```diff
@@ -56,6 +58,28 @@ def diff_as_string(actual, expected)
       end
       # rubocop:enable Metrics/MethodLength
 
+      def diff_hashes_as_object(actual, expected)
+        if defined?(RSpec::Mocks::ArgumentMatchers::AnyArgMatcher)
+          anything_hash = expected.select { |_, v| RSpec::Mocks::ArgumentMatchers::AnyArgMatcher === v }
+
+          anything_hash.each_key do |k|
+            expected[k] = actual[k]
+          end
+
+          diff_string = diff_as_object(actual, expected)
+
+          if defined?(RSpec::Mocks::ArgumentMatchers::AnyArgMatcher)
```

Comment:
Is this necessary? Looks like the same check on line 62

Reply:
Ooops... not necessary, I missed erasing that line. I've just submitted a commit with the removal.